server.go

```diff
@@ -288,6 +294,47 @@ func (s *Server) load() error {
 	})
 }
 
+//  StartRetentionPolicyEnforcement launches retention policy enforcement.
+func (s *Server) StartRetentionPolicyEnforcement(checkInterval time.Duration) error {
+	if checkInterval == 0 {
+		return fmt.Errorf("retention policy check interval must be non-zero")
+	}
+	rpDone := make(chan struct{}, 0)
+	s.rpDone = rpDone
+	go func() {
+		for {
+			select {
+			case <-rpDone:
+				return
+			case <-time.After(checkInterval):
+				s.EnforceRetentionPolicies()
+			}
+		}
+	}()
+	return nil
+}
+
+// EnforceRetentionPolicies ensures that data that is aging-out due to retention policies
+// is removed from the server.
+func (s *Server) EnforceRetentionPolicies() {
+	log.Println("retention policy enforcement check commencing")
+
+	// Check all shard groups.
+	for _, db := range s.databases {
+		for _, rp := range db.policies {
+			for _, g := range rp.shardGroups {
+				if g.EndTime.Add(rp.Duration).Before(time.Now()) {
```

Comment:
Should be using UTC here, otherwise it's going to be off if their server isn't set to that.